eng/Analyzers.targets

```
@@ -1,10 +1,13 @@
 <Project>
   <PropertyGroup>
     <!-- Disable analyzers in sourcebuild -->
-    <RunAnalyzers Condition="'$(DotNetBuildFromSource)' == 'true'">false</RunAnalyzers>
+    <RunAnalyzers Condition="'$(RunAnalyzers)' == '' And '$(DotNetBuildFromSource)' == 'true'">false</RunAnalyzers>
+    <EnableNETAnalyzers Condition="'$(EnableNETAnalyzers)' == ''">$(RunAnalyzers)</EnableNETAnalyzers>
   </PropertyGroup>
   <PropertyGroup Condition="'$(RunAnalyzers)' != 'false'">
-    <EnableSingleFileAnalyzer>true</EnableSingleFileAnalyzer>
+    <EnableSingleFileAnalyzer Condition="
+      '$(EnableSingleFileAnalyzer)' == '' And
+      $([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'netcoreapp3.0'))">true</EnableSingleFileAnalyzer>
```

We don't support building netcoreapp3.0 or anything below net6.0 in this repository. Given that, could we instead condition on the TargetFrameworkIdentifier which is just an msbuild property instead of calling the function (which has some overheard in msbuild evaluation)?

```suggestion
      '$(TargetFrameworkIdentifier)' == '.NETCoreApp'">true</EnableSingleFileAnalyzer>
```